specifyweb/frontend/js_src/lib/components/Forms/Save.tsx

```diff
@@ -204,17 +213,69 @@ export function SaveButton<SCHEMA extends AnySchema = AnySchema>({
       onClick={(): void => {
         // Scroll to the top of the form on clone
         smoothScroll(form, 0);
-        loading(handleClick().then(handleAdd));
+        loading(
+          handleClick()
+            .then((result) => {
+              const ids: WritableArray<number> = [];
+              if (originalResourceId !== undefined) {
+                ids.push(originalResourceId);
+              }
+              if (Array.isArray(result)) {
+                result.forEach((newResource) => {
+                  handleAdd?.(newResource);
+                  ids.push(newResource.id);
+                });
+              } else {
+                handleAdd?.(result);
+              }
+              return ids;
+            })
+            .then((ids) => {
+              if (handleCarryBulk && ids.length > 2) {
+                handleCarryBulk(ids);
+              }
+            })
+        );
       }}
     >
       {label}
     </ButtonComponent>
   );
 
+  const [carryForwardAmount, setCarryForwardAmount] = React.useState<number>(2);
+  const [showBulkCarryInput, setBulkCarryInput] = React.useState(false);
+
   return (
     <>
       {typeof handleAdd === 'function' && canCreate ? (
         <>
+          {resource.specifyTable.name === 'CollectionObject' &&
+          !isInRecordSet &&
+          !showBulkCarryInput &&
+          !resource.needsSaved ? (
+            <Button.Save
+              title={commonText.bulkCarry()}
+              onClick={() => setBulkCarryInput(true)}
+            >
+              {commonText.bulkCarry()}
+            </Button.Save>
+          ) : null}
+          {resource.specifyTable.name === 'CollectionObject' &&
+          !isInRecordSet &&
+          showBulkCarryInput ? (
+            <Label.Inline>
```

Comment:
should this be in a dialog?
otherwise, you would need to add aria-controls and forward focus from the button to input on button click

Reply:
not sure of the UI requirements for this PR. what do you think, @grantfitzsimmons ?

Reply:
@maxpatiiuk, is there usages of forward focus already in the code?

Reply:
A dialog would be the simplest if we can, as then all a11y works automatically.

Otherwise have to:
- Add `autofocus` attribute to the input
- Add this to the input: `forwardRef={(input)=>input?.focus()}`

you could look at these examples also, or search for more calls of `.focus()` in the codebase:
https://github.com/specify/specify7/blob/production/specifyweb/frontend/js_src/lib/components/FormCells/FormTable.tsx#L145
https://github.com/specify/specify7/blob/production/specifyweb/frontend/js_src/lib/components/Errors/Toasts.tsx#L89

Reply:
thank you!